src/components/CharacterDetails.tsx

```diff
@@ -0,0 +1,31 @@
+import React from 'react'
+import { isEmpty } from 'lodash'
+
+export type CharacterDetailsObject = {
+  bmi?: number
+  height?: number
+  mass?: number
+  homeworld?: string
+  born?: string
+  died?: string
+}
+
+type CharacterItemProps = {
+  details: CharacterDetailsObject
+}
+
+const CharacterDetails = ({ details }: CharacterItemProps) => {
+  if (isEmpty(details)) return <p>No detailed Data</p>
+
+  return (
+    <>
+      {details.homeworld && <li>Homeworld: {details.homeworld}</li>}
+      {details.born && <li>Year of birth: {details.born}</li>}
+      {details.died && <li>Dyear of death: {details.died}</li>}
+      {details.height && <li>Height: {details.height} m</li>}
+      {details.mass && <li>Weight: {details.mass} kg</li>}
+    </>
+  )
```

Comment:
Double ampersands(`&&`) operator is considered to be a bad practice in JSX due to its behavior. It acts undesired when the evaluated value is not `boolean` like when the value is `undefined`. I'd use ternary operator instead(value ? option1 : option2) 

reference: https://medium.com/geekculture/stop-using-for-conditional-rendering-in-react-a0f7b96200f8